ring/func/base.py

```diff
@@ -13,6 +14,11 @@
 from ..key import CallableKey
 from ..coder import registry as default_registry
 
+try:
+    from dataclasses import is_dataclass, asdict # noqa
+except ImportError:  # pragma: no cover
+    pass
```

Comment:
Then `dataclasses = None` will work to check the module exists or not at line 100.